cogs/strike.py

```diff
@@ -870,6 +870,121 @@ async def strike(self, ctx: "TeXBotApplicationContext", str_strike_member_id: st
+                str_user_id,
+            )
+        except ValueError as member_id_not_integer_error:
+            await self.command_send_error(ctx, message=member_id_not_integer_error.args[0])
+            return
+
+        strike_obj: DiscordMemberStrikes | None = None
+        try:
+            strike_obj = next(
+                strike_object
+                for strike_object in [
+                    all_strike_object
+                    async for all_strike_object in DiscordMemberStrikes.objects.select_related().all()  # noqa: E501
+                ]
+                if str(strike_object.discord_member)  # type: ignore[has-type]
+                == (DiscordMember.hash_discord_id(strike_member.id))
+            )
+        except StopIteration:
+            logger.debug("No strikes found for user %s", strike_member)
+
+        await ctx.respond(
+            content=(
+                f"User {strike_member.mention} has "
+                f"{strike_obj.strikes if strike_obj else 0} strikes."
+            ),
+            ephemeral=True,
+        )
+
+    @discord.slash_command(  # type: ignore[misc, no-untyped-call]
+        name="decrement-strikes",
+        description="Decrement the number of strikes a user has by 1.",
+    )
+    @discord.option(  # type: ignore[misc, no-untyped-call]
+        name="user",
+        description="The user to remove a strike from.",
+        input_type=str,
+        autocomplete=discord.utils.basic_autocomplete(autocomplete_get_members),  # type: ignore[arg-type]
+        required=True,
+        parameter_name="str_user_id",
+    )
+    async def decrement_strikes(  # type: ignore[misc]
+        self, ctx: "TeXBotApplicationContext", str_user_id: str
+    ) -> None:
+        """
+        Definition & callback response of the "decrement-strikes" command.
+
+        The "decrement-strikes" command removes a strike from the given member.
+        If the user only has one strike, the object will be deleted from the database.
+        """
+        try:
+            strike_member: discord.Member = await self.bot.get_member_from_str_id(
+                str_user_id,
+            )
+        except ValueError as member_id_not_integer_error:
+            await self.command_send_error(ctx, message=member_id_not_integer_error.args[0])
+            return
+
+        strike_obj: DiscordMemberStrikes | None = None
+        try:
+            strike_obj = next(
+                strike_object
+                for strike_object in [
+                    all_strike_object
+                    async for all_strike_object in DiscordMemberStrikes.objects.select_related().all()  # noqa: E501 # NOTE: I tried to reformat this to avoid the ruff error but the stupid fucking reformat put it all back on one line again
+                ]
+                if str(strike_object.discord_member)  # type: ignore[has-type]
+                == (DiscordMember.hash_discord_id(strike_member.id))
+            )
+        except StopIteration:
```

Comment:
Variable declaration before try-except is unnecessary, as the except branch returns from the function.
```suggestion
        try:
            discord_member_strikes: DiscordMemberStrikes = await DiscordMemberStrikes.objects.select_related().aget(
                discord_id=strike_member.id
            )
        except DiscordMemberStrikes.DoesNotExist:
```